### PR TITLE
 [Milestone 5.1] Implement Core Operational Transformation Data Structures and Utilities

### PR DESCRIPTION
**Pull request by EbenezerKobinaAcquah**
**Description:** 
--------------
This PR implements the foundational data structures and utility functions needed for Operational Transformation (OT) in the collaborative editor. 

**Changes:**

*   Added `constants/operationTypes.js` defining INSERT, DELETE, and RETAIN operations
*   Implemented `utils/operationUtils.js` with core functions:
    *   `createOperation()` for creating standardized operation objects
    *   `composeOps()` for combining sequential operations
    *   `invertOp()` for creating inverse operations (undo capability)
*   Added `utils/transform.js` with transformation functions:
    *   `transformOp()` for transforming operations against each other
    *   `transformPosition()` for adjusting cursor positions

**Why is this valuable?** 

This PR lays the groundwork for true real-time collaboration by implementing the core algorithms that will ensure consistency across multiple users editing simultaneously. The OT system will allow us to:

*   Maintain a consistent document state across all clients
*   Support concurrent editing without conflicts
*   Enable proper cursor tracking and visualization

**Resources and References**

[Basics of Operational Transformation](https://dev.to/sudoboink/the-basics-of-operational-transformation-288j)

**Test Plan:**

*   Tested with various operation combinations and edge cases
*   Verified mathematical properties of OT (convergence, intention preservation)

### CODE REVIEW

File: backend/utils/transform.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import { OP_RETAIN, OP_INSERT, OP_DELETE } from '../constants/operationTypes.js';
+import { normalizeOps, createRetainOp, createInsertOp, createDeleteOp } from './operationUtils.js';
+
+/**
+ * Transform operation A against operation B
+ * Returns A' such that: B + A' = A + B
+ */
```
Comment (vseth1027) [inline backend/utils/transform.js:7]: Good use of comments in this file! It really helps the reader follow along.

**Review by vseth1027 — APPROVED** [review]
Looks great! Very clean code and easy to understand.

**Review by evagarces — APPROVED** [review]
this was a nice set of changes! well written & organized!
